util/argo/argo.go

```diff
@@ -20,20 +28,53 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
-
-	argoappv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
-	"github.com/argoproj/argo-cd/v2/pkg/client/clientset/versioned/typed/application/v1alpha1"
-	applicationsv1 "github.com/argoproj/argo-cd/v2/pkg/client/listers/application/v1alpha1"
-	"github.com/argoproj/argo-cd/v2/reposerver/apiclient"
-	"github.com/argoproj/argo-cd/v2/util/db"
-	"github.com/argoproj/argo-cd/v2/util/io"
-	"github.com/argoproj/argo-cd/v2/util/settings"
 )
 
 const (
 	errDestinationMissing = "Destination server missing from app spec"
 )
 
+// FormatSyncMsg enrich the K8s message with user-relevant information
+func FormatSyncMsg(res *common.ResourceSyncResult, apiVersFn func() ([]kube.APIResourceInfo, error)) error {
+
+	switch res.Message {
+	case "the server could not find the requested resource":
+
+		resource, err := getAPIResourceInfo(res.ResourceKey.Kind, apiVersFn)
+		if err != nil {
+			return err
+		}
+		res.Message = fmt.Sprintf("The server could not find requested resource %s. Make sure the CRD is installed on the destination cluster, "+
+			"and that the requested CRD version is available. Currently, the installed API version for the corresponding Kind: %s is %s",
+			res.ResourceKey.Name, resource.GroupKind.Kind, resource.GroupVersionResource.Version)
+	}
+
+	return nil
+}
+
+func getAPIResourceInfo(kind string, apiVersFn func() ([]kube.APIResourceInfo, error)) (*kube.APIResourceInfo, error) {
+
+	var resource *kube.APIResourceInfo
+	apiResources, err := apiVersFn()
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range apiResources {
+		if r.GroupKind.Kind == kind {
+			resource = &r
+			break
+		}
+	}
```

Comment:
I think we should look for a version in a matching group first.

I think this will solve problems where `kind` names match, but aren't actually related.

For example, `argoproj.io/Application v1alpha2` should beat `otherproj.io/Application v1` when the input is `group=argoproj.io, kind=Application`.

```suggestion

	for _, r := range apiResources {
		if r.GroupKind.Group == group && r.GroupKind.Kind == kind {
			return &r, nil
		}
	}

	for _, r := range apiResources {
		if r.GroupKind.Kind == kind {
			return &r, nil
		}
	}
```